api/v1beta1/openstackdataplanerole_types.go

```diff
@@ -107,11 +107,21 @@ func (instance *OpenStackDataPlaneRole) InitConditions() {
 		condition.UnknownCondition(condition.DeploymentReadyCondition, condition.InitReason, condition.InitReason),
 		condition.UnknownCondition(condition.InputReadyCondition, condition.InitReason, condition.InitReason),
 		condition.UnknownCondition(SetupReadyCondition, condition.InitReason, condition.InitReason),
-		condition.UnknownCondition(RoleBareMetalProvisionReadyCondition, condition.InitReason, condition.InitReason),
-		condition.UnknownCondition(RoleIPReservationReadyCondition, condition.InitReason, condition.InitReason),
-		condition.UnknownCondition(RoleDNSDataReadyCondition, condition.InitReason, condition.InitReason),
 	)
 
+	// Only set Baremetal related conditions if we have baremetal hosts included in the
+	// baremetalSetTemplate.
+	if len(instance.Spec.BaremetalSetTemplate.BaremetalHosts) >= 1 {
+		bmConditionsList := []condition.Type{
+			RoleBareMetalProvisionReadyCondition,
+			RoleIPReservationReadyCondition,
+			RoleDNSDataReadyCondition,
+		}
```

Comment:
This list could probably go into the api package where these constants are defined. That might be a bit cleaner than having them here.